build/commands/scripts/format.js

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+const path = require('path')
+const fs = require('fs-extra')
+const prettier = require('prettier')
+const program = require('commander')
+
+const config = require('../lib/config')
+const util = require('../lib/util')
+
+// A function that formats the code in the current diff with base branch.
+// It uses git cl format and prettier, then aggregates the results.
+const runFormat = async (options = {}) => {
+  if (!options.base) {
+    options.base = 'origin/master'
+  }
+  let cmdOptions = config.defaultOptions
+  cmdOptions.cwd = config.braveCoreDir
+  cmdOptions = util.mergeWithDefault(cmdOptions)
+  cmd = 'git'
+  args = ['cl', 'format', '--upstream=' + options.base]
+
+  args.push('--python')
+  args.push('--no-rust-fmt')
+
+  if (options.full) args.push('--full')
+  if (options.diff) args.push('--diff')
```

Comment:
you need to add `--dry-run` support to `format.js` and also pass it from presubmit. `--diff` alone is not enough when checking for format errors.

Reply:
That makes sense. I would like to keep the number of configurations that format support as minimum as possible.
Currently we have
* "normal" format
* from presubmit 
* from presubmit --fix

What if we add `--presubmit` instead?

Reply:
> What if we add `--presubmit` instead?

that may work. In this case we should pass `--dry-run --presubmit --diff`.